From 565287c1ad8a6aafaeb22841b16adf0fb51ee960 Mon Sep 17 00:00:00 2001
Message-Id: <565287c1ad8a6aafaeb22841b16adf0fb51ee960.1327382345.git.dwm@austin.ibm.com>
From: Doug Maxey <dwm@austin.ibm.com>
Date: Mon, 23 Jan 2012 23:00:14 -0600
Subject: [PATCH] iw_cxgb4: workaround powerpc64 rping hang as a driver specific change
Patch-mainline: never
References: bnc#721857

Per the comment by mmarek@suse.com, this change disables the
ocqp_support in the driver itself conditionally based on
CONFIG_PPC64.

This issue was reported in LTCBZ 75462 - Novell721857.  The
issue has been confirmed by Chelsio development, this is just
another iteration to workaround the issue until the permanent
fix is implemented.

Signed-off-by: Doug Maxey <dwm@austin.ibm.com>
Cc: Divy Le Ray <divy@chelsio.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/infiniband/hw/cxgb4/qp.c |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/infiniband/hw/cxgb4/qp.c b/drivers/infiniband/hw/cxgb4/qp.c
index a41578e..049f55f 100644
--- a/drivers/infiniband/hw/cxgb4/qp.c
+++ b/drivers/infiniband/hw/cxgb4/qp.c
@@ -31,9 +31,15 @@
  */
 #include "iw_cxgb4.h"
 
-static int ocqp_support = 1;
+#ifdef CONFIG_PPC64
+#define IW_CXGB4_OCQP_SUPPORT_DEFAULT 0
+#else
+#define IW_CXGB4_OCQP_SUPPORT_DEFAULT 1
+#endif
+static int ocqp_support = IW_CXGB4_OCQP_SUPPORT_DEFAULT;
 module_param(ocqp_support, int, 0644);
-MODULE_PARM_DESC(ocqp_support, "Support on-chip SQs (default=1)");
+MODULE_PARM_DESC(ocqp_support, "Support on-chip SQs (default="
+		 __stringify(IW_CXGB4_OCQP_SUPPORT_DEFAULT) ")");
 
 static void set_state(struct c4iw_qp *qhp, enum c4iw_qp_state state)
 {
-- 
take care,
++doug

